src/req-manager.js

```diff
@@ -1,41 +1,65 @@
-var mod = require("module"),
-  path = require("path"),
-  fs = require("fs"),
-  logger = require("./logger");
+import fs from "node:fs";
+import mod from "node:module";
+import path from "node:path";
+import logger from "./logger.js";
 
-var Req = function (filepath, root) {
+const require = mod.createRequire(import.meta.url); // eslint-disable-line
+
+const Req = function(filepath, root) {
   this.Module = mod.Module;
   this.root = root;
   this.filepath = filepath || null;
   this.nativeModules = [
-    "assert", "buffer", "child_process", "constants", "crypto", "tls", "dgram", "dns", "http", "https",
-    "net", "querystring", "url", "domain", "events", "fs", "path", "os", "punycode", "stream", "string_decoder",
-    "timers", "tty", "util", "sys", "vm", "zlib"
+    "assert",
+    "buffer",
+    "child_process",
+    "constants",
+    "crypto",
+    "tls",
+    "dgram",
+    "dns",
+    "http",
+    "https",
+    "net",
+    "querystring",
+    "url",
+    "domain",
+    "events",
+    "fs",
+    "path",
+    "os",
+    "punycode",
+    "stream",
+    "string_decoder",
+    "timers",
+    "tty",
+    "util",
+    "sys",
+    "vm",
+    "zlib",
   ];
 };
 
-Req.prototype.load = function (modname) {
-  var self = this;
-
-  var cachedModule = self.Module._cache[self.filepath];
+Req.prototype.load = function(modname) {
+  const cachedModule = this.Module._cache[this.filepath];
   if (cachedModule) {
     return cachedModule.exports;
   }
 
-  return (function (self, mod, modname) {
+  return ((self, mod, modname) => {
     if (!modname.startsWith("./") && self.nativeModules.indexOf(modname) < 0) {
```

Comment:
This line seems kind of dodgy with respect to the above list. It just does `indexOf` rather than === which seems quite likely to create false positives/negatives given the generic names of most `node` modules (`fs`, `util`, `path`, `os`, `sys` etc) and the constantly moving set of node modules that exist. Do you understand the intent of this logic? It seems like there should be a better way.

Seems like "if module is not local and doesn't look like a built-in node module, normalise path", but this will _not_ normalise path for any other module which happens to have certain strings in it.

Reply:
This looks like some kind of module caching, but not sure how exactly it's used. But checking it against a list of hardcoded node js libraries does look wrong.

Reply:
Did a bit of digging. It's used at https://github.com/getgauge/gauge-js/blob/master/src/vm.js

The runtime is sandboxed to avoid access to internal methods. So I think these are helper methods for the sandboxed environment.

More context here  https://github.com/getgauge/gauge-js/issues/23

Reply:
I don't really understand the detail enough, my only concern is that this approach may or may no longer work as expected now it is wrapped with ES modules, or with `node:fs` rather than `fs` etc.

If that's not a sensible problem to worry about, can ignore. :-)

Reply:
I am not sure but I trust the e2e tests enough to catch regressions though!

Reply:
Yeah, how it interacts with people's random custom code/tests is the mystery, since the E2E tests are likely by definition opinionated as to how their example specs/step definitions are structured. As long as we are comfortable it covers real world expected usage it's probably fine.

Reply:
If you're comfortable with the coverage and risk here, feel free to merge at will. I'd feel more at ease if we had a plugin version pinning approach (ya, I know you can downgrade), but it's *probably* OK or can be quickly addressed afterwards if not? :-)

Reply:
Yeah. I'll keep and eye on issues and do a patch release.